cpp/ql/src/Security/CWE/CWE-311/CleartextFileWrite.ql

```diff
@@ -61,17 +78,12 @@ predicate isFileName(GVN gvn) {
 }
 
 from
-  FromSensitiveConfiguration config, SensitiveExpr source, DataFlow::PathNode sourceNode, Expr mid,
+  FromSensitiveConfiguration config, SensitiveExpr source, DataFlow::PathNode sourceNode,
   DataFlow::PathNode midNode, FileWrite w, Expr dest
 where
   config.hasFlowPath(sourceNode, midNode) and
-  sourceNode.getNode().asExpr() = source and
-  midNode.getNode().asExpr() = mid and
-  mid = w.getASource() and
-  dest = w.getDest() and
-  not dest.(VariableAccess).getTarget().getName() = ["stdin", "stdout", "stderr"] and // exclude calls with standard streams
-  not isFileName(globalValueNumber(source)) and // file names are not passwords
-  not exists(string convChar | convChar = w.getSourceConvChar(mid) | not convChar = ["s", "S"]) // ignore things written with other conversion characters
+  isSourceImpl(sourceNode.getNode(), source) and
+  isSinkImpl(midNode.getNode(), w, dest)
```

Comment:
This is cleaner than before the fix. :+1:

Reply:
Thanks! It should hopefully also be a bit faster since there'll be fewer sources and sinks considered by dataflow 🤞.